### PR TITLE
Fix includes being removed from count queries

### PR DESCRIPTION
**Pull request by BridgeAR**
### Pull Request check-list

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

See #6190 for more details (same fix applied against v.3)

### CODE REVIEW

File: test/integration/model/count.test.js [hunk 1]
```diff
@@ -133,5 +133,47 @@ describe(Support.getTestDialectTeaser('Model'), function() {
...
+      }).then((count) => {
+        expect(parseInt(count)).to.be.eql(0);
+      });
+    });
+
+    it('should have the same count result as findAndCount', function() {
+      let result;
+      const queryObject = {
+        include: [this.Project],
+        logging: function (sql) {
```
Comment (janmeier) [inline test/integration/model/count.test.js]: Whats going on with the logging here?

File: test/integration/model/count.test.js [hunk 2]
```diff
@@ -133,5 +133,47 @@ describe(Support.getTestDialectTeaser('Model'), function() {
...
+      queryObject.where['$Projects.name$'] = 'project1';
+      return this.User.count(queryObject).then((count) => {
+        expect(parseInt(count)).to.be.eql(1);
+        const queryObject = {
+          col: 'username',
+          include: [this.Project],
+          where: {}
+        };
+        queryObject.where['$Projects.name$'] = 'project2';
+        return this.User.count(queryObject);
```
Comment (janmeier) [inline test/integration/model/count.test.js]: No need to create the object first
```js
{
  col: 'username',
  include: [this.Project],
  where: {
    '$Projects.name$': 'project1'
  }
};
```

**Comment by janmeier** [comment]
Good to have you back @BridgeAR ! :)

File: test/integration/model/count.test.js (hunk 2 above)
Comment (BridgeAR) [inline test/integration/model/count.test.js]: :+1: I'm to much used to using variables in the names ;-)

File: test/integration/model/count.test.js (hunk 1 above)
Comment (BridgeAR) [inline test/integration/model/count.test.js]: The logging is used to compare the ouput of the `findAndCount` count query and the `count` query. They should always be identical with one minor exception: If you pass attributes in a findAndCount query the attributes should be stripped and in a count they should be kept.

Comment (janmeier) [inline test/integration/model/count.test.js]: Hmm, that feels a bit dirty 

You could also write this as a unit test, where you verify that the arguments to find and count are correct

File: test/integration/model/count.test.js (hunk 2 above)
Comment (BridgeAR) [inline test/integration/model/count.test.js]: Switching to eslint would be nice I guess. The rules there are much more powerful than the jshint ones. And maybe [standard](https://github.com/feross/standard) would also be a good idea to use.

Comment (janmeier) [inline test/integration/model/count.test.js]: Yep, eslint would be nice :) https://github.com/sequelize/sequelize/issues/5963

**Comment by BridgeAR** [comment]
Comments incorporated

File: test/unit/model/count.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  , Support = require(__dirname + '/../support')
+  , current = Support.sequelize
+  , sinon = require('sinon')
+  , DataTypes = require(__dirname + '/../../../lib/data-types')
+  , Promise = require('bluebird');
+
+describe(Support.getTestDialectTeaser('Model'), () => {
+  describe.only('method count', () => {
+    before(() => {
+      this.oldFindAll = current.Model.findAll;
```
Comment (janmeier) [inline test/unit/model/count.test.js]: You can preface with `sandbox = sinon.sandbox.create(), and use `sandbox.stub`. Then you just need to call `sandbox.restore()` in  ` afterEach` to reset all the stubs

**Comment by janmeier** [comment]
LGTM, would like @mickhansen or @sushantdhimans comments as well

File: test/integration/model.test.js [hunk 4]
```diff
@@ -1772,22 +1772,23 @@ describe(Support.getTestDialectTeaser('Model'), function() {
...
-              return Post.count({ include: [{ model: PostComment, required: false }] }).then(function(count1) {
-                return Post.count({ distinct: true, include: [{ model: PostComment, required: false }] }).then(function(count2) {
-                  expect(count1).to.equal(2);
-                  expect(count2).to.equal(1);
-                });
-              });
-            });
-          });
+      return Post.sync({ force: true }).then(() => {
+        return PostComment.sync({ force: true });
```
Comment (sushantdhiman) [inline test/integration/model.test.js]: Unnecessary returns in arrow functions, you can do ``` () => ( PostComment.sync({ force: true }) )```, Same applies for this whole chain

**Comment by sushantdhiman** [comment]
LGTM, I agree that performance check is causing too many issues, we should remove it for now.

Later on I would report if there are any significant deviation in performance.

**Comment by BridgeAR** [comment]
Done. I removed the distinct default too, as this depicts the regular database behavior.

**Comment by janmeier** [comment]
[No-no-no-NONET-kill!(https://en.wikipedia.org/wiki/Nonet_(music)) Pretty sure thats a new record for most issues fixed in one PR :)

**Pull request merged**